cdap-app-fabric/src/main/java/co/cask/cdap/security/hive/HiveTokenUtils.java

```diff
@@ -70,6 +79,65 @@ public static Credentials obtainToken(Credentials credentials) {
     }
   }
 
+  /**
+   * Builds a class loader with the class path provided.
+   */
+  public static ClassLoader getHiveClassloader() {
+    if (exploreClassLoader != null) {
```

Comment:
If the idea is to have a Singleton of the explore classLoader (which I think it make sense), then this method should be synchronized.